### ISSUE TITLE
Bug in at_hash verification in OpenID-Connect-PHP 3rd party library

### ISSUE THREAD

**Issue by coudot** [pulsejet/nextcloud-oidc-login#42]
Hello,

I found a little bug by using this plugin, the bug comes from OpenID-Connect-PHP 3rd party library. The verification of at_hash is not conform to OpenID Connect specification.

There is a pull request here for people facing the same issue: https://github.com/jumbojett/OpenID-Connect-PHP/pull/222

**Comment by azmeuk** [comment]
Are there visible consequences of this bug in nextcloud-oidc-login?

**Comment by coudot** [comment]
Yes of course, it fails with an OpenID Connect provider which respect the standard.

But it seems the bug was already fixed in https://github.com/jumbojett/OpenID-Connect-PHP/pull/200

So you just need to use a recent version of the lib.

**Comment by azmeuk** [comment]
The lib has been updated since the bug was fixed. Can you confirm this is fixed on the nextcloud-oidc-login side?

**Comment by coudot** [comment]
I don't have access to the server for now, I'll see if I can find time to do a new installation.
